libraries/lib-wave-track/WaveTrack.cpp

```diff
@@ -4201,198 +4233,18 @@ void WaveTrack::Resample(int rate, BasicUI::ProgressDialog *progress)
-               currentEnd = currentEnd -
-                  (clipEnd - clipStart) - (nextClipStart - clipEnd);
-            }
-
-            // detach the clip from track
-            revClips.push_back(track.RemoveAndReturnClip(clip));
-            // align time to a sample and set offset
-            revClips.back()->SetPlayStartTime(
-               track.SnapToSample(offsetStartTime));
-         }
-      }
-      else if (clipStart >= end) {
-         // clip is after the selection region
-         // simply remove and append to otherClips
-         otherClips.push_back(track.RemoveAndReturnClip(clip));
-      }
-   }
-
-   // STEP 3: Append the clips from
-   // revClips and otherClips back to the track
-   // the last clip of revClips is appended to the track first
-   // PRL:  I don't think that matters, the sequence of storage of clips in the
-   // track is not elsewhere assumed to be by time
-   for (auto it = revClips.rbegin(), revEnd = revClips.rend();
-        rValue && it != revEnd; ++it)
-      rValue = track.AddClip(*it);
-
-   if (!rValue)
-      return false;
-
-   for (auto &clip : otherClips)
-      if (!(rValue = track.AddClip(clip)))
-          break;
-
-   return rValue;
-}
-
-bool WaveTrack::ReverseOneClip(WaveTrack &track,
-   sampleCount start, sampleCount len,
-   sampleCount originalStart, sampleCount originalEnd,
-   const ProgressReport &report)
-{
-   bool rc = true;
-   // keep track of two blocks whose data we will swap
-   auto first = start;
-
-   auto blockSize = track.GetMaxBlockSize();
-   Floats buffer1{ blockSize };
-   const auto pBuffer1 = buffer1.get();
-   Floats buffer2{ blockSize };
-   const auto pBuffer2 = buffer2.get();
-
-   auto originalLen = originalEnd - originalStart;
-
-   while (len > 1) {
-      auto block =
-         limitSampleBufferSize(track.GetBestBlockSize(first), len / 2);
-      auto second = first + (len - block);
-
-      track.GetFloats(buffer1.get(), first, block);
-      std::reverse(pBuffer1, pBuffer1 + block);
-      track.GetFloats(buffer2.get(), second, block);
-      std::reverse(pBuffer2, pBuffer2 + block);
-      // Don't dither on later rendering if only reversing samples
-      const bool success =
-         track.Set((samplePtr)buffer2.get(), floatSample, first, block,
-            narrowestSampleFormat)
-         &&
-         track.Set((samplePtr)buffer1.get(), floatSample, second, block,
-            narrowestSampleFormat);
-      if (!success)
-         return false;
-
-      len -= 2 * block;
-      first += block;
-
-      if (!report(
-         2 * (first - originalStart).as_double() / originalLen.as_double()
-      )) {
-         rc = false;
-         break;
-      }
+   bool result = true;
+   size_t ii = 0;
+   for (const auto &pChannel : Channels()) {
+      const auto buffer = buffers[ii++];
+      assert(buffer); // precondition
+      result = pChannel->SetFloats(buffer, start, len, effectiveFormat)
```

Comment:
I think you meant to use an `&=` (not `|=`), or this method will always return true.

Reply:
OK
With just that change, I assume your tacit reapproval